hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -127,11 +129,20 @@ class WorkflowParams(param.Parameterized):
     This class contains all parameters that affect how the whole training and testing workflow is executed.
     """
     random_seed: int = param.Integer(42, doc="The seed to use for all random number generators.")
-    weights_url: str = param.String(default="",
-                                    doc="If provided, use this URL to download checkpoints for inference.")
-    local_weights_path: Optional[Path] = \
-        param.ClassSelector(class_=Path, allow_None=True, default=None,
-                            doc="Checkpoint paths to use for inference, when the job is running outside Azure.")
+    src_checkpoint: str = param.String(default="",
+                                       doc="This flag can be used for 3 different scenarios:"
+                                           "1- Resume training from a checkpoint to train longer."
+                                           "2- Run inference only by jointly using `run_inference_only` flag."
+                                           "3- Transfer learning from a pretrained model checkpoint."
+                                           "We currently support Three types of checkpoints: "
+                                           "    a. A local checkpoint folder that contains a checkpoint file."
+                                           "    b. A URL to a remote checkpoint to be downloaded."
+                                           "    c. A previous azureml run id where the checkpoint is supposed to be "
+                                           "       saved in 'outputs/checkpoints/' folder.")
+    src_checkpoint_filename: str = param.String(default="last.ckpt",
```

Comment:
I would add that to the Azure run ID via a `:`.
Also, we probably want to be more flexible with where the checkpoint can live. For example, if I just say `foo:last.ckpt` it will search for it in `outputs/checkpoints`. But I should also be able to say `foo:some/other/path.ckpt`

Reply:
Done!